src/main/java/com/gengo/client/GengoClient.java

```diff
@@ -352,9 +364,147 @@ public JSONObject getTranslationJobs() throws GengoException
      */
     public JSONObject getTranslationJobs(List<Integer> ids) throws GengoException
     {
-        String url = baseUrl + "translate/jobs/";
-        url += join(ids, ",");
-        return call(url, HttpMethod.GET);
+        return this.getTranslationJobs(ids, null, null, null);
+    }
+
+    /**
+     * Get specific translation jobs.
+     * @param ids a list of job ids to retrieve
+     * @param status one of the JobStatus
+     * @param timestampAfter Unix epoch seconds from which to filter submitted jobs
+     * @param count limit count of jobs(default 10, maximum 200)
+     * @return
+     * @throws GengoException
+     */
+    public JSONObject getTranslationJobs(List<Integer> ids, JobStatus status, Integer timestampAfter, Integer count) throws GengoException
+    {
+        String url = this.getBaseUrl() + "translate/jobs/";
+        JSONObject data = new JSONObject();
+        if (ids != null && ids.size() > 0) {
+            url += join(ids, ",");
+        }
+        if (status != null) {
+            try {
+                data.put("status", status);
+            } catch (JSONException x) {
+                throw new GengoException(x.getMessage(), x);
+            }
+        }
+        if (timestampAfter != null && timestampAfter > 0) {
+            try {
+                data.put("timestamp_after", timestampAfter);
+            } catch (JSONException x) {
+                throw new GengoException(x.getMessage(), x);
+            }
+        }
+        if (count != null && count > 0) {
+            try {
+                data.put("count", count);
+            } catch (JSONException x) {
+                throw new GengoException(x.getMessage(), x);
+            }
+        }
+        return call(url, HttpMethod.GET, data);
+    }
+
+    /**
+     * Set revision comment for each jobs.
+     * @param jobs each Entry consists of job ID and comment text
+     * @return
+     * @throws GengoException
+     */
+    public JSONObject reviseTranslationJobs(HashMap<Integer, String> jobs) throws GengoException
+    {
+        try
+        {
+            String url = this.getBaseUrl() + "translate/jobs";
+            JSONObject data = new JSONObject();
+            data.put("action", "revise");
+            //[begin] Generate 'job_ids' parameter.
+            JSONArray job_ids = new JSONArray();
+            for (Entry<Integer, String> iJob : jobs.entrySet()) {
+                HashMap<String, Object> job = new HashMap<String, Object>();
+                job.put("job_id", iJob.getKey());
+                job.put("comment", iJob.getValue());
+                job_ids.put(job);
+            }
+            //[end] Generate 'job_ids' parameter.
+            data.put("job_ids", job_ids);
+            return call(url, HttpMethod.PUT, data);
+        } catch (JSONException x)
```

Comment:
Maybe you want to check your own code before the question. https://github.com/gengo/gengo-java/blob/master/src/main/java/com/gengo/client/GengoClient.java#L138

Anyway, I'll update those variables in the file. Thanks.